core/src/main/java/org/web3j/protocol/websocket/WebSocketService.java

```diff
@@ -128,11 +128,10 @@ private void connectToWebSocket() throws InterruptedException, ConnectException
                         ? webSocketClient.reconnectBlocking()
                         : webSocketClient.connectBlocking();
 
+        shouldReConnect = true;
```

Comment:
I would add a functionality to make some limited attempts of web socket connection. (webSocketClient.connectBlocking())

Reply:
> I would add a functionality to make some limited attempts of web socket connection. (webSocketClient.connectBlocking())

I don't understand. `connect`which calls `connectToWebsocket` is a public API, so the external user could decide whether and when he wants to reconnect. You want to make such logic to be internal?

Reply:
@andrii-kl added test, which fails in a current master

Reply:
@zilm13 I guess @andrii-kl wanted to say to set up a counter in order to establish a connection pool. I took a look at code and I don't think is necessary, even more there is the onClose() method that is called on every listener.
Your test makes sense. 👍